file/random_access_file_reader.h

```diff
@@ -100,8 +102,13 @@ class RandomAccessFileReader {
 #endif
   }
 
-  RandomAccessFileReader(RandomAccessFileReader&& o) ROCKSDB_NOEXCEPT {
-    *this = std::move(o);
+  RandomAccessFileReader(RandomAccessFileReader&& o) ROCKSDB_NOEXCEPT
+      : file_(std::move(o.file_)) {
```

Comment:
What's the reason for ```std::move``` of individual member variables instead of ```std::move(o)```? Also, what about ```file_name_``` and ```listeners_```?

Reply:
In std::move(o), it expects  FSRandomAccessFilePtr to have default constructor, which  in turns expect FSRandomAccessFileTracingWrapper(since its a object) to have default Constructor but FSRandomAccessFileTracingWrapper is inherited from FSRandomAccessFileWrapper and it expects pointer to file system (No default constructor). So I added it to do member by member move. (similar to move operator). But i wanted to remove these move functions as operator= also doesn't have ```file_name_``` and ```listeners_``` .